lib/rspec/support/caller_filter.rb

```diff
@@ -25,7 +25,7 @@ class CallerFilter
     # when `CallerFilter.first_non_rspec_line` is called from the top level of a required
     # file, but it depends on if rubygems is loaded or not. We don't want to have to deal
     # with this complexity in our `RSpec.deprecate` calls, so we ignore it here.
-    IGNORE_REGEX = Regexp.union(LIB_REGEX, "rubygems/core_ext/kernel_require.rb")
+    IGNORE_REGEX = Regexp.union(LIB_REGEX, "rubygems/core_ext/kernel_require.rb", "<internal:")
```

Comment:
This could be more precise/efficient, with `/^<internal:/` to only look if it starts with that.

Reply:
@eregon I've updated to the recommended `/^<internal:/`